core/src/main/java/io/cloudevents/core/impl/CloudEventReaderAdapter.java

```diff
@@ -25,8 +25,11 @@
 
```

Comment:
You can improve this even further and remove boilerplate subclassing `CloudEventContextReaderAdapter`:

```java
public class CloudEventReaderAdapter extends CloudEventContextReaderAdapter implements CloudEventReader
```

Reply:
I prefer composition over inheritance but it's your choice (236fff0).